jerry-core/include/jerryscript-port.h

```diff
@@ -138,6 +138,39 @@ double jerry_port_get_current_time (void);
  */
 struct jerry_instance_t *jerry_port_get_current_instance (void);
 
+/*
+ * Debugger Port API
+ */
+
+/* forward declaration */
+struct jerry_debugger_uint8_data_t;
+
+/**
+ * Transport APIs, different transport will define
+ * the real implementation of these APIs
+ */
+struct jerry_debugger_transport_t
+{
+  uint8_t send_header_size;
+  uint8_t receive_header_size;
+  uint8_t max_message_size;
+  bool (*accept_connection) (struct jerry_debugger_transport_t *transport_p);
+  void (*close_connection) (struct jerry_debugger_transport_t *transport_p);
+  bool (*send) (struct jerry_debugger_transport_t *transport_p,
+                uint8_t *message_data_p,
+                size_t data_size);
+  bool (*receive) (struct jerry_debugger_transport_t *transport_p,
+                   uint8_t *message_data_p,
+                   size_t *data_size,
+                   uint32_t *buffer_offset);
+};
+
+/**
+ * The default transport layer that uses TCP socket
+ * other transports can be added like usb, bluetooth
+ */
+struct jerry_debugger_transport_t *jerry_port_init_socket_transport (uint16_t tcp_port);
+
```

Comment:
I'm not 100% sure whether this should be in `jerryscript-port.h` because it's specific to an implementation of a transport (of a TCP socket transport in this case).

This is comparable/similar to the various `jerry_port_default_...` functions in https://github.com/jerryscript-project/jerryscript/blob/master/jerry-port/default/include/jerryscript-port-default.h

Reply:
@martijnthe thanks for catching that, i updated patch